extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -189,12 +190,15 @@ export class AmpAnalytics extends AMP.BaseElement {
             trigger['selector'] = this.expandTemplate_(trigger['selector'],
                 trigger, /* arg*/ undefined, /* arg */ undefined,
                 /* arg*/ false);
-            this.instrumentation_.addListener(
-                trigger, this.handleEvent_.bind(this, trigger), this.element);
-
+            this.instrumentationPromise_.then(instrumentation => {
```

Comment:
Yes this would delay the layout for `amp-analytics` but it might make sense in this case since I guess analytics functionality is dependent on instrumentation being ready.